default/python/AI/InvasionAI.py

```diff
@@ -327,14 +327,13 @@ def _get_path_from_capital(planet: "fo.planet") -> Tuple[Sequence[SystemId], int
     """
     universe = fo.getUniverse()
     capitol_id = PlanetUtilsAI.get_capital()
-    if capitol_id:
+    if capitol_id != INVALID_ID:
```

Comment:
don't also need to test for `None`?

Reply:
Argh, these checks are a pain in the ass, some functions return None, some INVALID_ID, perhaps some do both. At the moment get_capital() does not return None, but I agree its safer to check both.

And there is still the question whether 0 is a valid id, because the "standard check" `if id and id != INVALID_ID` also yields false if id is 0

Reply:
There won't normally be a planet with id 0. Might be a system 0, though.

Reply:
Actually yes, there is a system with ID 0. Just had a test game in which one AI started in it. It never left the system. Haven't yet figured out why. Log says:

`21:08:28.839817 {0x00007fa656944780} [warn] python : target.py:19 : Target is invalid inaccessible system_-1<>`
`21:08:28.839890 {0x00007fa656944780} [debug] python : MoveUtilsAI.py:51 : Requesting path for fleet F_3902<Recon Fleet 3902> from inaccessible sys>`
`21:08:28.839993 {0x00007fa656944780} [warn] python : pathfinding.py:204 : Requested path between invalid systems.`

But this is a separate issue...